include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -979,6 +1033,270 @@ namespace internal
+        std::vector<dealii::Vector<double>> &patch_values_system)
+        const override;
+
+      /**
+       * Given a FEValuesBase object, extract the gradients on the present
+       * cell from the vector we actually store.
+       */
+      virtual void
+      get_function_gradients(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<Tensor<1, DoFHandlerType::space_dimension>>
+          & /*patch_gradients*/) const override
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+      /**
+       * Given a FEValuesBase object, extract the gradients on the present
+       * cell from the vector we actually store. This function does the same
+       * as the one above but for vector-valued finite elements.
+       */
+      virtual void
+      get_function_gradients(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<std::vector<Tensor<1, DoFHandlerType::space_dimension>>>
+          & /*patch_gradients_system*/) const override
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+
+      /**
+       * Given a FEValuesBase object, extract the second derivatives on the
+       * present cell from the vector we actually store.
+       */
+      virtual void
+      get_function_hessians(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<Tensor<2, DoFHandlerType::space_dimension>>
+          & /*patch_hessians*/) const override
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+      /**
+       * Given a FEValuesBase object, extract the second derivatives on the
+       * present cell from the vector we actually store. This function does
+       * the same as the one above but for vector-valued finite elements.
+       */
+      virtual void
+      get_function_hessians(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<std::vector<Tensor<2, DoFHandlerType::space_dimension>>>
+          & /*patch_hessians_system*/) const override
+      {
+        Assert(false, ExcNotImplemented());
+      }
+
+      /**
+       * Return whether the data represented by (a derived class of) this object
+       * represents a complex-valued (as opposed to real-valued) information.
+       */
+      virtual bool
+      is_complex_valued() const override
+      {
+        return false;
```

Comment:
that's not correct -- you should query what the vector type actually does. (You may not be in a position right now to make use of complex-valued values, but you're introducing a difficult-to-find bug if you just return `false` here.)

Reply:
You are right, fixed.